File: 696-count-binary-substrings/count-binary-substrings.cpp
class Solution {
public:
    int countBinarySubstrings(string s) {
        vector<int>v;
        int cnt=1;
        for(int i=1;i<s.size();i++)
        {
            if(s[i]==s[i-1])
            cnt++;
            else
            {
                v.push_back(cnt);
                cnt=1;
            }
        }
        v.push_back(cnt);
        int ans=0;
        for(int i=1;i<v.size();i++)
        {
            ans+=min(v[i],v[i-1]);

        }
        return ans;
        
    }
};

File: 696-count-binary-substrings/count-binary-substrings.py
class Solution:
    def countBinarySubstrings(self, s: str) -> int:
        n=len(s)
        val=[]
        cnt_0=1
        cnt_1=1
        for i in range(1,n):
            if s[i]==s[i-1]:
                if s[i]=='0':
                    cnt_0+=1
                else:
                    cnt_1+=1
            else:
                val.append(max(cnt_0,cnt_1))
                cnt_0=1
                cnt_1=1
        
        val.append(max(cnt_0,cnt_1))
        mp=defaultdict(int)
        mp[0]=1
        cnt=0
        ans=0
        for i in range(1,len(val)):
            cnt+=min(val[i],val[i-1])
        return cnt

            


        